### PR TITLE
Switch to KernelAbstractions

### LINKED ISSUE #10: Use extensions for different backends
See: https://github.com/SciML/DiffEqGPU.jl/blob/master/ext/CUDAExt.jl

### PR DESCRIPTION
**Pull request by jpsamaroo**
Resolves #10 

Todo is to also setup CI to test this

### CODE REVIEW

File: .buildkite/runtests.yml [hunk 1]
```diff
@@ -26,6 +26,34 @@ steps:
...
+    plugins:
+      - JuliaCI/julia#v1:
+          version: "{{matrix.version}}"
+    commands:
+      - |
+        julia --project=test -e '
+          import Pkg
+          Pkg.add(; name = "AMDGPU")'
+        rm test/Manifest.toml
+        julia --project -e'
```
Comment (Vaibhavdixit02) [inline .buildkite/runtests.yml:45]: Why does it say Test not found if we activate in the main directory in the error, does this need to instantiate here?

File: src/ode_pso.jl [hunk 2]
```diff
@@ -1,47 +1,43 @@
...
         c2 = 1.4962f0)
-    i = (blockIdx().x - 1) * blockDim().x + threadIdx().x
-    i > length(gpu_particles) && return
+    i = @index(Global, Linear)
+    if i <= length(gpu_particles)
+        @inbounds particle = gpu_particles[i]
 
-    @inbounds particle = gpu_particles[i]
+        updated_velocity = w .* particle.velocity .+
+                           c1 .* rand(typeof(particle.velocity)) .* (particle.best_position -
```
Comment (Vaibhavdixit02) [inline src/ode_pso.jl:8]: Should be `randn`

Comment (jpsamaroo) [reply src/ode_pso.jl:8]: This, and the one below it? Which other such calls might need changing?

Comment (Vaibhavdixit02) [reply src/ode_pso.jl:8]: Actually, let's do it in a separate PR. There's some code repetition so it will have to be changed in a couple of more places.

**Comment by jpsamaroo** [comment]
Yeah I'm pretty happy with its current state I think

**Pull request merged**